src/modules/check_test.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2019, Circonus, Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ *       copyright notice, this list of conditions and the following
+ *       disclaimer in the documentation and/or other materials provided
+ *       with the distribution.
+ *     * Neither the name Circonus, Inc. nor the names
+ *       of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written
+ *       permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef CHECK_TEST_H
+#define CHECK_TEST_H
+
+#include <mtev_hooks.h>
+
+MTEV_RUNTIME_AVAIL(noit_testcheck_lookup,
+                   noit_testcheck_lookup_dyn)
+MTEV_RUNTIME_RESOLVE(noit_testcheck_lookup,
+                     noit_testcheck_lookup_dyn,
+                     noit_check_t *,
+                     (uuid_t in),
+                     (in))
+
+#define NOIT_TESTCHECK_LOOKUP(noit_check_ptr,uuid) do { \
+  if(noit_testcheck_lookup_available()) { \
+    noit_check_ptr = noit_testcheck_lookup(uuid); \
+  } \
+  else { \
+    noit_check_ptr = NULL; \
+  } \
+} while(0)
```

Comment:
I am not a fan of global macros that treat args are lvalues -- it's very un C.

```
#define NOIT_TESTCHECK_LOOKUP(uuid) \
  (noit_testcheck_lookup_available() ? noit_testcheck_lookup(uuid) : (noit_check_t *)NULL)
```

Reply:
will commit this, sounds good